c-ares 1.33.1 fails to build with googletest 1.15.2

On my machine, trying to build c-ares version 1.33.1 gives compiler errors:
```
In file included from ../../../test/ares-test-mock.cc:26:
../../../test/ares-test.h:799:16: error: expected ‘;’ at end of member declaration
  799 |     static int gtest_registering_dummy_ GTEST_ATTRIBUTE_UNUSED_;               \
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
../../../test/ares-test-mock.cc:836:1: note: in expansion of macro ‘CONTAINED_TEST_P’
  836 | CONTAINED_TEST_P(ContainedMockChannelSysConfig, SysConfigNdotsDefault,
      | ^~~~~~~~~~~~~~~~
../../../test/ares-test.h:799:41: error: ‘GTEST_ATTRIBUTE_UNUSED_’ does not name a type; did you mean ‘GTEST_ATTRIBUTE_PRINTF_’?
  799 |     static int gtest_registering_dummy_ GTEST_ATTRIBUTE_UNUSED_;               \
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~
../../../test/ares-test-mock.cc:836:1: note: in expansion of macro ‘CONTAINED_TEST_P’
  836 | CONTAINED_TEST_P(ContainedMockChannelSysConfig, SysConfigNdotsDefault,
      | ^~~~~~~~~~~~~~~~
../../../test/ares-test.h:799:16: error: expected ‘;’ at end of member declaration
  799 |     static int gtest_registering_dummy_ GTEST_ATTRIBUTE_UNUSED_;               \
      |                ^~~~~~~~~~~~~~~~~~~~~~~~
../../../test/ares-test-mock.cc:861:1: note: in expansion of macro ‘CONTAINED_TEST_P’
  861 | CONTAINED_TEST_P(ContainedMockChannelSysConfig, SysConfigNdots0,
      | ^~~~~~~~~~~~~~~~
../../../test/ares-test.h:799:41: error: ‘GTEST_ATTRIBUTE_UNUSED_’ does not name a type; did you mean ‘GTEST_ATTRIBUTE_PRINTF_’?
  799 |     static int gtest_registering_dummy_ GTEST_ATTRIBUTE_UNUSED_;               \
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~
../../../test/ares-test-mock.cc:861:1: note: in expansion of macro ‘CONTAINED_TEST_P’
  861 | CONTAINED_TEST_P(ContainedMockChannelSysConfig, SysConfigNdots0,
      | ^~~~~~~~~~~~~~~~
make[2]: *** [Makefile:1151: arestest-ares-test-mock.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory '/home/unfrobbed-dev/pkgs/c-ares/c-ares-1.33.1/debian/obj-amd64/test'
make[1]: *** [Makefile:541: all-recursive] Error 1
make[1]: Leaving directory '/home/unfrobbed-dev/pkgs/c-ares/c-ares-1.33.1/debian/obj-amd64'
```

On the other hand, c-ares version 1.33.0 built successfully with the same version of googletest.

--- a/test/ares-test.h
+++ b/test/ares-test.h
@@ -768,6 +768,17 @@ int RunInContainer(ContainerFilesystem *fs, const std::string &hostname,
     }                                                                         \
     int ICLASS_NAME(casename, testname)::InnerTestBody()
 
+
+
+/* Derived from googletest/include/gtest/gtest-param-test.h, specifically the
+ * TEST_P() macro, and some fixes to try to be compatible with different
+ * versions. */
+#ifndef GTEST_ATTRIBUTE_UNUSED_
+#  define GTEST_ATTRIBUTE_UNUSED_
+#endif
+#ifndef GTEST_INTERNAL_ATTRIBUTE_MAYBE_UNUSED
+#  define GTEST_INTERNAL_ATTRIBUTE_MAYBE_UNUSED
+#endif
 #  define CONTAINED_TEST_P(test_suite_name, test_name, hostname, domainname, \
                            files)                                            \
     class GTEST_TEST_CLASS_NAME_(test_suite_name, test_name)                 \
@@ -781,9 +792,9 @@ int RunInContainer(ContainerFilesystem *fs, const std::string &hostname,
       {                                                                      \
         ContainerFilesystem chroot(files, "..");                             \
         VoidToIntFn         fn = [this](void) -> int {                       \
-          ares_reinit(this->channel_);                               \
-          ares_sleep_time(100);                                      \
-          return this->InnerTestBody();                              \
+          ares_reinit(this->channel_);                                       \
+          ares_sleep_time(100);                                              \
+          return this->InnerTestBody();                                      \
         };                                                                   \
         EXPECT_EQ(0, RunInContainer(&chroot, hostname, domainname, fn));     \
       }                                                                      \
@@ -803,12 +814,8 @@ int RunInContainer(ContainerFilesystem *fs, const std::string &hostname,
             ::testing::internal::CodeLocation(__FILE__, __LINE__));          \
         return 0;                                                            \
       }                                                                      \
-      static int gtest_registering_dummy_ GTEST_ATTRIBUTE_UNUSED_;           \
-      GTEST_TEST_CLASS_NAME_(test_suite_name, test_name)                     \
-      (const GTEST_TEST_CLASS_NAME_(test_suite_name, test_name) &) = delete; \
-      GTEST_TEST_CLASS_NAME_(test_suite_name, test_name) &operator=(         \
-        const GTEST_TEST_CLASS_NAME_(test_suite_name,                        \
-                                     test_name) &) = delete; /* NOLINT */    \
+      GTEST_INTERNAL_ATTRIBUTE_MAYBE_UNUSED static int                       \
+        gtest_registering_dummy_ GTEST_ATTRIBUTE_UNUSED_;                    \
     };                                                                       \
     int GTEST_TEST_CLASS_NAME_(test_suite_name,                              \
                                test_name)::gtest_registering_dummy_ =        \

